django/db/models/expressions.py

```diff
@@ -483,7 +485,14 @@ def __init__(self, value, output_field=None):
         self.value = value
 
     def as_sql(self, compiler, connection):
-        return '%s', [self.value]
+        val = self.value
+        if self._output_field_or_none is not None:
+            from django.db.models.sql.compiler import SQLUpdateCompiler
+            if isinstance(compiler, SQLUpdateCompiler):
```

Comment:
Not liking this. The expression should not need to know what kind of compiler is being used here. Why is this necessary here, but hasn't been required before? Is this something that resolve_ref can handle without full knowledge of the type of compiler? Or do we need to re-add a similar get_db_prep method into expressions?